yarn-project/foundation/src/abi/buffer.ts

```diff
@@ -0,0 +1,36 @@
+import chunk from 'lodash.chunk';
+
+import { Fr } from '../fields/fields.js';
+
+/**
+ * Formats a buffer as an array of fields. Splits the input into 31-byte chunks, and stores each
+ * of them into a field, omitting the field's first byte, then adds zero-fields at the end until the max length.
+ * @param input - Input to format.
+ * @param targetLength - Length of the target array in number of fields.
+ * @returns A field with the total length in bytes, followed by an array of fields such that their concatenation is equal to the input buffer, followed by enough zeroes to reach targetLength.
+ */
+export function bufferAsFields(input: Buffer, targetLength: number): Fr[] {
+  const encoded = [
+    new Fr(input.length),
+    ...chunk(input, Fr.SIZE_IN_BYTES - 1).map(c => {
+      const fieldBytes = Buffer.alloc(32);
```

Comment:
Should this be `Buffer.alloc(Fr.SIZE_IN_BYTES)` ? Also, do we only write 31 byte chunks to ensure we don't overflow the size of a field element?

Reply:
> Should this be Buffer.alloc(Fr.SIZE_IN_BYTES) ? 

Yep, you're right, I'll change in an upcoming PR.

> Also, do we only write 31 byte chunks to ensure we don't overflow the size of a field element?

That's correct. A field element fits between 31 and 32 bytes.